src/library.cpp

```diff
@@ -79,6 +79,38 @@ static void ptr_argument_warning()
   ptr_argument_flag = 0;
 }
 
+// for throwing exceptions from within the library interface; similar to Error::all()
+// instead of qualifying it with source file and line, we use the (library) function name
+// __func__ is supposed to be portable for compilers also supporting C99: MSVC, GCC, Clang.
+#if defined(FNERR)
+#undef FNERR
+#endif
+#define FNERR __func__
+
+[[noreturn]] static void lammps_throw_error(const std::string &fname, const std::string &mesg)
+{
+  throw LAMMPSException("ERROR in library function " + fname + "(): " + mesg + "\n");
+  exit(1);
+}
+
+[[noreturn]] static void _lammps_throw_error(const std::string &fname, fmt::string_view format,
+                                      fmt::format_args args)
+{
+  try {
+    lammps_throw_error(fname, fmt::vformat(format, args));
+  } catch (fmt::format_error &e) {
+    lammps_throw_error(fname, e.what());
+  }
+  exit(1); // to trick "smart" compilers into believing this does not return
+}
+
+template <typename... ARgs>
```

Comment:
ARgs is weird. Just call it Args.

Reply:
turns out this code is not needed at all.